test/unit/lodash.spec.js

```diff
@@ -0,0 +1,1326 @@
+        });
+        it('should return new object with defaults properties', function(){
+            var defaultsResults = _.defaultsDeep.apply(_, params);
+            expect(defaultsResults).to.have.property('data');
+            expect(defaultsResults).to.have.deep.property('data.text', 'Tim');
+            expect(defaultsResults).not.to.have.deep.property('data.text', 'mouse');
+            expect(defaultsResults).to.have.deep.property('data.reversedText', 'miT');
+            expect(defaultsResults).not.to.have.deep.property('data.reversedText', 'esuom');
+            expect(defaultsResults).to.have.deep.property('data.lengthText');
+        });
+    });
+
+    describe('findKey', function () {
+        describe('findKey1', function () {
+            var params;
+            before(function(){
+                params = datasets.findKey1();
+            });
+            it('should check types', function(){
+                var elem2 = params[1];
+                expect(elem2 instanceof Function).to.eql(true);
+            });
+            it('should return person who has number greather than 20', function(){
+                var obj1 = {};
+
+                //function _.times Returns the array. In this case 100 results.
+                _.times(100, function(index) {
+
+                    //function _.random Returns the random number between 0 and 20
+                    obj1['key' + index] = {number: _.random(20)};
+                });
+                //function _.random Returns the random number between 0 and 99
+                var expectedKey = _.keys(obj1)[_.random(99)];
+
+                //function _.random Returns the random number between 0 and 10
+                obj1[expectedKey]={number:21+ _.random(10)};
+
+                expect(_.findKey.apply(_, datasets.findKey1(obj1))).to.eql(expectedKey);
+            });
+        });
+
+        describe('findKey2', function () {
+            var params;
+            before(function(){
+                params = datasets.findKey2();
+            });
+            it('should check types', function(){
+                var elem2 = params[1];
+                expect(typeof elem2).to.eql('object');
+            });
+            it('should return keys with value { number: 15 }', function(){
+                var obj1 = {};
+
+                //function _.times Returns the array. In this case 100 results.
+                _.times(100, function(index) {
+
+                    //function _.random Returns the random number between 0 and 20
+                    obj1['key' + index] = {number: _.random(20)};
+                });
+
+                //function _.random Returns the random number between 0 and 99
+                var expectedKey = _.keys(obj1)[_.random(99)];
+
+                //function _.random Returns the random number between 0 and 10
+                obj1[expectedKey] = {number: 21 + _.random(10)};
+
+                var newObject = obj1;
```

Comment:
copying reference of obj1 to newObject is redundant here and in many other places.